public/app/core/internationalization/dates.ts

```diff
@@ -41,3 +41,19 @@ export const formatDuration = deepMemoize(
     return dateFormatter.format(duration);
   }
 );
+
+export const formatRange = (range: string, options: Intl.DurationFormatOptions = {}): string => {
+  // Split the date using regex to get the "to" word, even translated, to get the "from" and "to" values
+  const regex = /([a-zA-Z]+)/g;
+  const timeRangeSplit = range.split(regex);
+  const from = timeRangeSplit[0];
+  const to = timeRangeSplit[2];
+  // If from and to are not empty, localise them using the formatDate function
+  if (from && to) {
+    const fromLocalised = formatDate(from, options);
+    const toLocalised = formatDate(to, options);
+    const separator = timeRangeSplit[1];
+    range = `${fromLocalised} ${separator} ${toLocalised}`;
+  }
+  return range;
+};
```

Comment:
i don't think we should be regex splitting a string. you could do this `formatDate` stuff when the string is constructed in the `mapRangeToTimeOption` function here: https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker/mapper.ts#L7